packages/react/src/useFindBy.ts

```diff
@@ -89,6 +91,12 @@ export const useFindBy = <
       ],
     });
   }
+  const nonNullableError = getNonNullableError(result, dataPath);
```

Comment:
For the existing implementation: I'm not sure why I didn't build this on `useFindMany`, with an appropriately constructed set of options (which is roughly how it's implemented in api-client-core). Have we duplicated a lot of (tricky) work in here, along with some of the new hooks being introduced?

Reply:
So I spent a bit of time looking at this, and:
1. There's an underlying `findOneByFieldOperation` operation, which _does_ compose `findManyOperation`. That said, this operation has a different signature than the `findMany` one, and I don't think it makes sense to avoid using it here — I think that would make for more fragile code, if anything.
2. For the `data` and `error` processing, there is a lot of code overlap, but there are also some slight differences.

Given that, I'm included to not compose these two hooks. I think the most we could do is cleanup the code in all these hooks by having a util for generating the `useQuery` object — it seems to be the same in all hooks.

I'll add that in, because it means more focus on the important logic when reading these bits of code, but you guys can let me know if you're against it.